src/initialize.cpp

```diff
@@ -0,0 +1,226 @@
+        openmc_run_mode = RUN_MODE_PLOTTING;
+        openmc_check_overlaps = true;
+
+      } else if (arg == "-n" || arg == "--particles") {
+        i += 1;
+        n_particles = std::stoll(argv[i]);
+
+      } else if (arg == "-r" || arg == "--restart") {
+        i += 1;
+
+        // Check what type of file this is
+        hid_t file_id = file_open(argv[i], 'r', true);
+        size_t len = attribute_typesize(file_id, "filetype");
+        read_attr_string(file_id, "filetype", len, buffer);
+        file_close(file_id);
+        std::string filetype {buffer};
+
+        // Set path and flag for type of run
+        if (filetype == "statepoint") {
+          openmc_path_statepoint = argv[i];
+          openmc_restart_run = true;
+        } else if (filetype == "particle restart") {
+          openmc_path_particle_restart = argv[i];
+          openmc_particle_restart_run = true;
+        } else {
+          std::stringstream msg;
+          msg << "Unrecognized file after restart flag: " << filetype << ".";
+          strcpy(openmc_err_msg, msg.str().c_str());
+          return OPENMC_E_INVALID_ARGUMENT;
+        }
+
+        // If its a restart run check for additional source file
+        if (openmc_restart_run && i + 1 < argc) {
+          // Check if it has extension we can read
+          if (ends_with(argv[i+1], ".h5")) {
+
+            // Check file type is a source file
+            file_id = file_open(argv[i+1], 'r', true);
+            len = attribute_typesize(file_id, "filetype");
+            read_attr_string(file_id, "filetype", len, buffer);
+            file_close(file_id);
+            if (filetype != "source") {
+              std::string msg {"Second file after restart flag must be a source file"};
+              strcpy(openmc_err_msg, msg.c_str());
+              return OPENMC_E_INVALID_ARGUMENT;
+            }
+
+            // It is a source file
+            openmc_path_sourcepoint = argv[i+1];
+            i += 1;
+
+          } else {
+            // Source is in statepoint file
+            openmc_path_sourcepoint = openmc_path_statepoint;
+          }
+
+        } else {
+          // Source is assumed to be in statepoint file
+          openmc_path_sourcepoint = openmc_path_statepoint;
+        }
+
+      } else if (arg == "-g" || arg == "--geometry-debug") {
+        openmc_check_overlaps = true;
+      } else if (arg == "-c" || arg == "--volume") {
+        openmc_run_mode = RUN_MODE_VOLUME;
+      } else if (arg == "-s" || arg == "--threads") {
+        // Read number of threads
+        i += 1;
+
+#ifdef _OPENMP
+        // Read and set number of OpenMP threads
+        openmc_n_threads = std::stoi(argv[i]);
+        if (openmc_n_threads < 1) {
+          std::string msg {"Number of threads must be positive."};
+          strcpy(openmc_err_msg, msg.c_str());
+          return OPENMC_E_INVALID_ARGUMENT;
+        }
+        omp_set_num_threads(openmc_n_threads);
+#endif
```

Comment:
We should probably output a warning message here if OpenMC wasn't built with OpenMPI.